src/python/pants/backend/docker/goals/run_image.py

```diff
@@ -3,32 +3,38 @@
 from __future__ import annotations
 
 import sys
-from typing import cast
+from typing import Iterator, cast
 
 from pants.backend.docker.goals.package_image import BuiltDockerImage, DockerFieldSet
+from pants.backend.docker.subsystems.docker_options import DockerOptions
 from pants.backend.docker.util_rules.docker_binary import DockerBinary
 from pants.core.goals.package import BuiltPackage, PackageFieldSet
 from pants.core.goals.run import RunRequest
-from pants.engine.rules import Get, collect_rules, rule
+from pants.engine.environment import Environment, EnvironmentRequest
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+
+
+def get_docker_run_args(options: DockerOptions) -> Iterator[str]:
+    if sys.stdout.isatty():
+        yield "-it"
```

Comment:
Do docker args provide a method for disabling `-it` (`-no-it`?) in case we detected a TTY incorrectly?

Reply:
Hmm.. no, I don't think it does. (It would be `--no-interactive` and `--no-tty`, but still)

I was thinking a bit about how to best handle this. And perhaps not add any default args at all, iff there are args `--docker-run-args` provided.. ?

Reply:
Maybe...? If folks are familiar with running docker directly, they might already expect to need to pass certain things?

Reply:
Yeah.. maybe, probably. Esp. given that there's no way to opt out of it..

Reply:
I'll try setting it as default instead, that way, if you provide your own, you can choose to add or replace the default, using standard option semantics. :)